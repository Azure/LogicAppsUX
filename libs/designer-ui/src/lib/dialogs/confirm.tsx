import { DefaultButton, PrimaryButton } from '@fluentui/react/lib/Button';
import { Dialog, DialogFooter, IDialogContentProps } from '@fluentui/react/lib/Dialog';
import { IModalProps } from '@fluentui/react/lib/Modal';
import * as React from 'react';
import { FormattedMessage } from 'react-intl';

export interface ConfirmProps {
  hidden: boolean;
  message: string;
  title: string;
  onConfirm(): void;
  onDismiss(): void;
}

const modalProps: IModalProps = {
  isBlocking: true,
  firstFocusableSelector: 'dialog-ok-button',
};

export const Confirm = ({ hidden, message, title, onConfirm, onDismiss }: ConfirmProps): JSX.Element | null => {
  if (hidden) {
    return null;
  }

  const dialogContentProps: IDialogContentProps = {
    title,
  };

  return (
    <Dialog dialogContentProps={dialogContentProps} hidden={hidden} modalProps={modalProps} onDismiss={onDismiss}>
      {message}
      <DialogFooter>
        <PrimaryButton className="dialog-ok-button" onClick={onConfirm}>
          <FormattedMessage defaultMessage="OK" description="OK message appearing on a confirmation dialog." />
        </PrimaryButton>
        <DefaultButton onClick={onDismiss}>
          <FormattedMessage defaultMessage="Cancel" description="Cancel message appearing on a confirmation dialog." />
        </DefaultButton>
      </DialogFooter>
    </Dialog>
  );
};
