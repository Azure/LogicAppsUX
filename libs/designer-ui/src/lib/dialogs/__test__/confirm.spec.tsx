import type { ConfirmProps } from '../confirm';
import { Confirm } from '../confirm';
import * as React from 'react';
import * as ReactShallowRenderer from 'react-test-renderer/shallow';
import { describe, vi, beforeEach, afterEach, beforeAll, afterAll, it, test, expect } from 'vitest';
describe('ui/dialogs/_confirm', () => {
  let minimal: ConfirmProps;
  let renderer: ReactShallowRenderer.ShallowRenderer;

  beforeEach(() => {
    minimal = {
      hidden: false,
      message: 'Message',
      title: 'Title',
      onConfirm: vi.fn(),
      onDismiss: vi.fn(),
    };
    renderer = ReactShallowRenderer.createRenderer();
  });

  it('should render', () => {
    const confirm = renderer.render(<Confirm {...minimal} />);

    expect(confirm).toMatchSnapshot();
  });
});
