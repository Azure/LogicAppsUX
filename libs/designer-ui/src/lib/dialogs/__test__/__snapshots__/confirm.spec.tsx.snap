// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/dialogs/_confirm should render 1`] = `
<Dialog
  dialogContentProps={
    {
      "title": "Title",
    }
  }
  hidden={false}
  modalProps={
    {
      "firstFocusableSelector": "dialog-ok-button",
      "isBlocking": true,
    }
  }
  onDismiss={[MockFunction]}
>
  Message
  <StyledDialogFooterBase>
    <CustomizedPrimaryButton
      className="dialog-ok-button"
      onClick={[MockFunction]}
    >
      OK
    </CustomizedPrimaryButton>
    <CustomizedDefaultButton
      onClick={[MockFunction]}
    >
      Cancel
    </CustomizedDefaultButton>
  </StyledDialogFooterBase>
</Dialog>
`;
